pennylane/ops/meta.py

```diff
@@ -167,6 +168,9 @@ class Snapshot(Operation):
         tag (str or None): An optional custom tag for the snapshot, used to index it
             in the snapshots dictionary.
 
+        measurement (StateMeasurement or None): An optional option to record arbitrary
```

Comment:
Are we restricted to state-based measurement processes only?

Reply:
For now at least. We might be able to open it up to sample-based measurements in the future, but that would require a few more changes.